### PR TITLE
ambient occlusion preference

### PR DESCRIPTION
**Pull request by harryob**
## About The Pull Request

adds a faint dropshadow around The World, our lighting engine looks less obviously bad + is stored as a preference so soul enjoyers can have their original version. same ambient occlusion as used on TG 😨

## Why It's Good For The Game

improves the aesthetics with a new look that makes full bright less intense and partially darkened rooms much nicer

https://user-images.githubusercontent.com/55142896/195763651-df0dd022-f6e0-4d3f-9fe2-5d07f08e4360.mp4

https://user-images.githubusercontent.com/55142896/195763679-21c64508-4131-4773-87a6-5623d56ba6af.mp4

## Changelog

:cl:
add: Ambient occlusion is now a(n enabled by default) preference.
/:cl:

### CODE REVIEW

File: code/_onclick/hud/rendering/plane_master.dm [hunk 1]
```diff
@@ -30,13 +30,25 @@
...
 /atom/movable/screen/plane_master/game_world
 	name = "game world plane master"
 	plane = GAME_PLANE
 	appearance_flags = PLANE_MASTER //should use client color
 	blend_mode = BLEND_OVERLAY
 
+/atom/movable/screen/plane_master/game_world/backdrop(mob/mymob)
+	. = ..()
+	remove_filter("AO")
+	if(istype(mymob) && mymob.client.prefs.ambientocclusion)
```
Comment (stanalbatross) [inline code/_onclick/hud/rendering/plane_master.dm:49]: sanity checks for client please

File: code/game/objects/items/explosives/plastic.dm [hunk 2]
```diff
@@ -167,7 +167,7 @@
 	update_icon()
 
 /obj/item/explosive/plastic/proc/can_place(var/mob/user, var/atom/target)
-	if(istype(target, /obj/structure/ladder) || istype(target, /obj/item) || istype(target, /turf/open) || istype(target, /obj/structure/barricade) || istype(target, /obj/structure/closet/crate))
+	if(istype(target, /obj/structure/ladder) || istype(target, /obj/item) || istype(target, /turf/open) || istype(target, /obj/structure/barricade) || isanimal(target))
```
Comment (stanalbatross) [inline code/game/objects/items/explosives/plastic.dm:170]: ?

File: code/modules/client/preferences.dm [hunk 3]
```diff
@@ -1600,6 +1603,13 @@ var/const/MAX_SAVE_SLOTS = 10
 					var/flag = text2num(href_list["flag"])
 					toggles_ert ^= flag
 
+				if("ambientocclusion")
```
Comment (stanalbatross) [inline code/modules/client/preferences.dm:1606]: consider a toggle verb in prefs too

File: code/modules/client/preferences_savefile.dm [hunk 4]
```diff
@@ -213,6 +214,7 @@
 	key_bindings 		= sanitize_keybindings(key_bindings)
 	remembered_key_bindings = sanitize_islist(remembered_key_bindings, null)
 	hotkeys  			= sanitize_integer(hotkeys, FALSE, TRUE, TRUE)
+	ambientocclusion	= sanitize_integer(ambientocclusion, 0, 1, initial(ambientocclusion))
```
Comment (stanalbatross) [inline code/modules/client/preferences_savefile.dm:217]: FALSE/TRUE please

**Review by stanalbatross — CHANGES_REQUESTED** [review]

**Comment by Wintermote** [comment]
Any performance impact/testing on performance?

**Comment by harryob** [comment]
> Any performance impact/testing on performance?

it shouldn't, due to the filter being applied to a planemaster. if someone wishes to testmerge this, i can make a branch that doesn't mess with prefs and just stores the preference on the client temporarily

File: code/_onclick/hud/rendering/plane_master.dm [hunk 5]
```diff
@@ -30,13 +30,25 @@
...
 /atom/movable/screen/plane_master/game_world
 	name = "game world plane master"
 	plane = GAME_PLANE
 	appearance_flags = PLANE_MASTER //should use client color
 	blend_mode = BLEND_OVERLAY
 
+/atom/movable/screen/plane_master/game_world/backdrop(mob/mymob)
+	. = ..()
+	remove_filter("AO")
+	if(istype(mymob) && mymob?.client.prefs.toggle_prefs & TOGGLE_AMBIENT_OCCLUSION)
```
Comment (Geevies) [inline code/_onclick/hud/rendering/plane_master.dm:49]: confirm that client and prefs exists

**Review by Geevies — CHANGES_REQUESTED** [review]

File: code/_onclick/hud/rendering/plane_master.dm (hunk 5 above)
Comment (harryob) [reply code/_onclick/hud/rendering/plane_master.dm:49]: i love question marks

**Pull request merged**